packages/react/src/context/slash-id-context.tsx

```diff
@@ -180,11 +226,32 @@ export const SlashIDProvider = ({
     async ({ oid: newOid }: { oid: string }) => {
       if (!user) return;
 
-      const newToken = await user.getTokenForOrganization(newOid);
+      setOrgSwitchingState({
+        state: "switching",
+        oid,
+      });
+
+      let newToken: string;
+
+      // check if a valid token for new org is in storage
+      const newOidToken = storageRef.current?.getItem(
+        `@slashid/USER_TOKEN/${newOid}`
+      );
+      const isNewOidTokenValid =
+        newOidToken && (await validateToken(newOidToken));
```

Comment:
Along with validating the token, should we also decode it and check the org ID to ensure it really is a token for the given org? Not sure if that would break some behaviour though 🤔 for example how a token for the root dashboard org can be used for suborgs too. Probably better to skip that check 🤔